commercial/app/model/commercial/Merchandise.scala

```diff
@@ -0,0 +1,370 @@
+                offerPrice: Option[Double] = None,
+                description: Option[String] = None,
+                jacketUrl: Option[String],
+                buyUrl: Option[String] = None,
+                position: Option[Int] = None,
+                category: Option[String] = None,
+                keywordIdSuffixes: Seq[String] = Nil) extends Merchandise
+
+case class Masterclass(id: String,
+                       name: String,
+                       startDate: DateTime,
+                       url: String,
+                       description: String,
+                       status: String,
+                       venue: EBVenue,
+                       tickets: Seq[EBTicket],
+                       capacity: Int,
+                       guardianUrl: String,
+                       firstParagraph: String,
+                       keywordIdSuffixes: Seq[String],
+                       mainPicture: Option[ImageElement]) extends Merchandise with EBTicketHandler with EBEventHandler {
+
+  lazy val readableDate = DateTimeFormat.forPattern("d MMMMM yyyy").print(startDate)
+
+  lazy val truncatedFirstParagraph = StringUtils.abbreviate(firstParagraph, 250)
+}
+
+case class LiveEvent(eventId: String,
+                     name: String,
+                     date: DateTime,
+                     eventUrl: String,
+                     description: String,
+                     status: String,
+                     venue: EBVenue,
+                     tickets: Seq[EBTicket],
+                     imageUrl: String) extends Merchandise with EBTicketHandler with EBEventHandler
+
+case class TravelOffer(id: String,
+                       title: String,
+                       offerUrl: String,
+                       imageUrl: String,
+                       fromPrice: Option[Double],
+                       earliestDeparture: DateTime,
+                       keywordIdSuffixes: Seq[String],
+                       countries: Seq[String],
+                       category: Option[String],
+                       tags: Seq[String],
+                       duration: Option[Int],
+                       position: Int) extends Merchandise {
+
+  val durationInWords: String = duration match {
+    case Some(1) => "1 night"
+    case Some(x) => s"$x nights"
+    case None => ""
+  }
+
+  def formattedPrice : Option[String] = fromPrice map { price =>
+    if (price % 1 == 0)
+      f"£$price%,.0f"
+    else
+      f"£$price%,.2f"
+  }
+}
+
+case class Member(username: String, gender: Gender, age: Int, profilePhoto: String, location: String) {
+
+  val profileId: Option[String] = profilePhoto match {
+    case Member.IdPattern(id) => Some(id)
+    case _ => None
+  }
+
+  val profileUrl: String = profileId.map(id => s"https://soulmates.theguardian.com/landing/$id")
+    .getOrElse("http://soulmates.theguardian.com/")
+
+}
+
+case class MemberPair(member1: Member, member2: Member) extends Merchandise
+
+sealed case class Gender(name: String) {
```

Comment:
Can we move this to the location of the things that `extend` it?

Also it's worth considering why you need `name` here as a parameter; I think you probably don't need it. You might even just be able to get away with `sealed trait Gender`, because you usually know the type if you are matching it out, unless you are using this base type directly and don't care if it's `Woman` or `Man`.